llvm/test/CodeGen/AIE/aie2/schedule/interblock/liveness-region.mir

```diff
@@ -0,0 +1,248 @@
+
+  bb.0:
+    liveins: $bml0, $bml1, $r0
+    successors: %bb.1
+    $bml2 = VADD_F $bml0, $bml1, $r0, implicit-def $srfpflags,  implicit $crfpmask
+    JL @callee, implicit-def $lr
+    DelayedSchedBarrier csr_aie2, implicit-def $r0
+    $r1 = MOV_mv_scl $r0
+  bb.1:
+    liveins: $bml0, $bml1, $bml2, $r1
+    successors: %bb.1, %bb.2
+    $bml3 = VADD_F $bml0, $bml2, $r1, implicit-def $srfpflags,  implicit $crfpmask
+    JL @callee, implicit-def $lr
+    DelayedSchedBarrier csr_aie2, implicit-def $bml1, implicit-def $bml3, implicit-def $r1
+    $bml1 = VADD_F $bml1, $bml3, $r1, implicit-def $srfpflags,  implicit $crfpmask
+    JL @callee, implicit-def $lr
+    DelayedSchedBarrier csr_aie2, implicit-def $bml1, implicit-def $bml2, implicit-def $r1
+    $bml2 = VADD_F $bml1, $bml2, $r1, implicit-def $srfpflags,  implicit $crfpmask
+    $r1 = ADD_add_r_ri $r1, -1, implicit-def $srcarry
+    JNZ $r1, %bb.1
+    DelayedSchedBarrier
+  bb.2:
+    liveins: $bml2
+    RET implicit $lr
+    DelayedSchedBarrier implicit $bml2
+...
+
+#     bb.0
+#       |
+#     bb.1-------*
+#       |        |
+#     bb.2-----* |
+#     /  \     | |
+# bb.3    bb.4 | |
+#     \  /     | |
+#     bb.5-----* |
+#       |        |
+#     bb.6-------*
+#       |
+#     bb.7
+---
+name:            liveness_nested_loop_regions
+alignment:       16
+tracksRegLiveness: true
+body:             |
+  ; CHECK-LIVENESS-LABEL: liveness_nested_loop_regions:BB0 LiveOuts
+  ; CHECK-LIVENESS-NEXT:   Live Registers: $r0 $r1
+  ; CHECK-LIVENESS-NEXT: liveness_nested_loop_regions:BB1 LiveOuts
+  ; CHECK-LIVENESS-NEXT:   Live Registers: $r2 $r4 $srcarry $r3
+  ; CHECK-LIVENESS-NEXT: liveness_nested_loop_regions:BB2 LiveOuts
+  ; CHECK-LIVENESS-NEXT:   Live Registers: $r2 $r4 $srcarry
+  ; CHECK-LIVENESS-NEXT: liveness_nested_loop_regions:BB3 LiveOuts
+  ; CHECK-LIVENESS-NEXT:   Live Registers: $srcarry
+  ; CHECK-LIVENESS-NEXT: liveness_nested_loop_regions:BB4 LiveOuts
+  ; CHECK-LIVENESS-NEXT:   Live Registers: $srcarry
+  ; CHECK-LIVENESS-NEXT: liveness_nested_loop_regions:BB5 LiveOuts
+  ; CHECK-LIVENESS-NEXT:   Live Registers: $r2 $r4 $srcarry $r3 $r0 $r1 $r9 $r8
+  ; CHECK-LIVENESS-NEXT: liveness_nested_loop_regions:BB6 LiveOuts
+  ; CHECK-LIVENESS-NEXT:   Live Registers: $r0 $r1 $r3 $r9 $srcarry
+  ; CHECK-LIVENESS-NEXT: liveness_nested_loop_regions:BB7 LiveOuts
+  ; CHECK-LIVENESS-NEXT:   Live Registers: (empty)
+  
+  bb.0:
+    successors: %bb.1
+    liveins: $r1, $r2
+    $r0 = nsw ADD $r1, $r2, implicit-def $srcarry
+    $r2 = MUL_mul_r_rr $r1, $r1
+    JL @callee, implicit-def $lr
+    DelayedSchedBarrier csr_aie2, implicit-def $r1
+    $r1 = ABS $r1, implicit-def $srcarry
+  bb.1:
+    successors: %bb.2
+    liveins: $r0, $r1
+    $r3 = nsw ADD $r0, $r1, implicit-def $srcarry
+    JL @callee, implicit-def $lr
+    DelayedSchedBarrier csr_aie2, implicit-def $r1, implicit-def $r3
+    $r1 = AND $r1, $r3
+  bb.2:
+    successors: %bb.3, %bb.4
+    liveins: $r2, $r3
```

Comment:
When we come from `bb.5`, I think `$r2` is actually undefined. Maybe extend `DelayedSchedBarrier csr_aie2, implicit-def $r6` to also define `$r2`? (And maybe others, I didn't check all the liveins, but the MachineVerifier should 😉)